numpy/random/mtrand.pyx

```diff
@@ -807,9 +807,11 @@ cdef class RandomState:
                 cdf /= cdf[-1]
                 uniform_samples = self.random_sample(shape)
                 idx = cdf.searchsorted(uniform_samples, side='right')
-                idx = np.array(idx, copy=False)  # searchsorted returns a scalar
+                # searchsorted returns a scalar
+                # force cast to int for LLP64
+                idx = np.array(idx, copy=False).astype(np.int, casting='unsafe') 
```

Comment:
`np.int` should be deprecated, you should use `int` directly

Reply:
Thanks, fixed.